method.rb

```diff
@@ -0,0 +1,99 @@
+module Enumerable
+  def my_each
+    (0...length).each do |i|
+      yield self[i]
+    end
+    self
+  end
+
+  def my_each_with_index
+    (0...length).each do |i|
+      yield self[i], i
+    end
+    self
+  end
+
+  def my_select
+    new_arr = []
+    my_each do |num|
+      new_arr.push(num) if yield num
+    end
+    new_arr
+  end
+
+  def my_all?
+    stat = true
+    my_each do |num|
+      stat = false unless yield num
+    end
+    stat
+  end
```

Comment:
- [ ] should have working `my_all?` *identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

- Your method doesn't consider the possibility when no block or argument is given. It should return true if the enumerable contains truthy values and false if it contains at least one falsy value.

- Your method doesn't consider the possibility when a class is given as an argument `(some_enumerable.my_all?(String)).`

- Your method doesn't consider the possibility when a Regex is given as an argument `(some_enumerable.my_all?(/w/)).`

- Your method doesn't consider the possibility when a number is given as an argument `(some_enumerable.my_all?(5)).`